src/map.cpp

```diff
@@ -630,45 +625,42 @@ bool Map::isSightClear(const Position& fromPos, const Position& toPos, bool floo
 			return true;
 		}
 
-		//check if tiles above us and target are clear and check for clear sight between them
+		//check if tiles above us and the target are clear and check for a clear sight between them
 		--z0;
 		return checkSightLine(x0, y0, x1, y1, z0) && isTileClear(x0, y0, z0, true) && isTileClear(x1, y1, z0, true);
-	} else {
-		//target is on a different floor
-		if (floorCheck) {
-			return false;
-		}
+	}
 
-		//skip checks for sight line in case fromPos and toPos cross the ground floor
-		if (z0 < 8 && z1 > 7 || z0 > 7 && z1 < 8) {
-			return false;
-		}
+	//target is on a different floor
+	if (floorCheck) {
+		return false;
+	}
 
-		if (z0 > z1) {
-			//target is above us
-			if (Position::getDistanceZ(fromPos, toPos) > 1) {
-				return false;
-			}
+	//skip checks for sight line in case fromPos and toPos cross the ground floor
+	if (z0 < 8 && z1 > 7 || z0 > 7 && z1 < 8) {
+		return false;
+	}
```

Comment:
This mechanic should not be here, since this check is only done to throw items
What this function of verifying the line of sight gives me to understand is knowing exactly that, if the line of sight is clean

Reply:
If you call isSightClear with sameFloor = false from lua script, it doesn't go through Map::canThrowObjectTo. This prevents iterating over path when there is no reason to do it.